extract_from_bibliography.py

```diff
@@ -18,16 +18,18 @@ def get_keys(filename: Path | str) -> Set[str]:
 def get_items_from_bib(keys: Iterable[str], filename: Path | str) -> Iterable[Tuple[str, str]]:
     with open(filename, 'r') as f:
         contents = f.read()
+    # We do our best but, in the end, regexes are too weak too parse bibtex.
     regex = re.compile(
         r"""
-        ^@ \w+ \{ (?P<key>%s),$\n          # KEY
-          .*?                    # BODY
-        ^\s*\}$""" % '|'.join(map(re.escape, keys)),
-        re.MULTILINE | re.VERBOSE | re.DOTALL)
+        (^|\n) \s* (?P<entry> @ \w+ \{ \s* (?P<key>%s) \s* ,  # KEY
+        .*?                                                   # BODY
+        \} ) (?= \s* @ \w+ \{ | $ )                           # END
+        """ % '|'.join(map(re.escape, keys)),
+        re.ASCII | re.VERBOSE | re.DOTALL)
```

Comment:
I'm not sure if `re.ASCII` is smart and/or necessary. I also am not sure if removing `re.MULTILINE` makes things easier. Without `re.MULTILINE`, you require that all entries are sequential with no comments in-between.